src/Pages/Overview/Overview.js

```diff
@@ -34,15 +43,48 @@ export default class OverviewBoard extends Component {
     if (!apiResponse.error) this.setState({ users: apiResponse.responseData })
   }
 
-  updateQuery (event) {
-    let { value } = event.target
+  updateQuery (value) {
+    // taking care of empty values
+    value = typeof value === 'undefined' ? '' : value
     value = value.trim().toLowerCase()
-    this.setState({ query: value })
+    let queryResult
+
+    const userExists = user => {
+      return (
+        user.firstName.toLowerCase().includes(value) ||
+        user.lastName.toLowerCase().includes(value) ||
+        user.email.toLowerCase().includes(value)
+      )
+    }
+
+    if (this.state.currentQueryType === 'All') {
```

Comment:
Now would be a good time to use a switch statement. [if else vs switch](https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#if-else_versus_switch)

```js
const { currentQueryType } = this.state

switch(currentQueryType) {
  case 'Officer':
    // do your thing
    break;
  /* etc here */
  default:
    // handle the 'All' case here
}
```